Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/connected_components.h

```diff
@@ -956,6 +962,180 @@ void keep_connected_components(PolygonMesh& pmesh
 }
 
+
+namespace internal{
+
+//overloads used to select a default index map:
+// use the one passed in the named parameters (user must have initialized it)
+template <class MapFromNP, class Default_tag, class Dynamic_tag, class Mesh, class Range>
+MapFromNP
+get_index_map(MapFromNP m, Default_tag, Dynamic_tag, const Mesh&, const Range&)
+{
+  return m;
+}
+
+// use the one internal to the mesh (user must have initialized it)
+template <class Default_tag, class Dynamic_tag, class Mesh, class Range>
+typename boost::property_map<Mesh, Default_tag >::const_type
+get_index_map(CGAL::internal_np::Param_not_found, Default_tag t, Dynamic_tag , const Mesh& m, const Range& )
+{
+
+  return get(t,m);
+}
+
+// create a dynamic property and initialize it
+template <class Dynamic_tag, class Mesh, class Range>
+typename boost::property_map<Mesh, Dynamic_tag >::const_type
+get_index_map(CGAL::internal_np::Param_not_found, Dynamic_tag t, Dynamic_tag , const Mesh& m, const Range& r)
+{
+  typename boost::property_map<Mesh, Dynamic_tag >::const_type map = get(t,m);
+
+  std::size_t id =0;
+  for(auto primitive : r)
+  {
+    put(map, primitive, id++);
+  }
+
+  return map;
+}
+
+template <typename G>
+struct No_mark
+{
+  friend bool get(No_mark<G>,
+                  typename boost::graph_traits<G>::edge_descriptor)
+  {
+    return false;
+  }
+  friend void put(No_mark<G>,
+                  typename boost::graph_traits<G>::edge_descriptor, bool)
+  {}
+};
+
+
+template < class PolygonMesh, class PolygonMeshRange,
+           class FIMap, class VIMap,
+           class HIMap, class Ecm >
+void split_connected_components_impl(
+    FIMap fim,
+    HIMap him,
+    VIMap vim,
+    Ecm ecm,
+    PolygonMeshRange& range,
+    const PolygonMesh& tm
+    )
+{
+  typename boost::template property_map<
+      PolygonMesh, CGAL::dynamic_face_property_t<int > >::const_type
+      pidmap = get(CGAL::dynamic_face_property_t<int>(), tm);
+
+  int nb_patches =
+      CGAL::Polygon_mesh_processing::connected_components(
+        tm,
+        pidmap,
+        CGAL::parameters::face_index_map(fim)
+        .edge_is_constrained_map(ecm));
+
+  for(int i=0; i<nb_patches; ++i)
+  {
+    CGAL::Face_filtered_graph<PolygonMesh, FIMap, VIMap, HIMap>
+        filter_graph(tm, i, pidmap,
+                     CGAL::parameters::face_index_map(fim)
+                     .halfedge_index_map(him)
+                     .vertex_index_map(vim));
+    range.push_back(PolygonMesh());
+    PolygonMesh& new_graph = range.back();
+    CGAL::copy_face_graph(filter_graph, new_graph);
+  }
+}
+}//internal
+
+/*!
+ * \ingroup keep_connected_components_grp
+ * identifies the connected components of `pm` and pushes back a new `PolygonMesh` for each connected component in `cc_meshes`.
+ *
+ *
+ *  \tparam PolygonMesh a model of `FaceListGraph`
+ *  \tparam PolygonMeshRange a model of `SequenceContainer` (providing
+ *  `void push_back()` and `PolygonMesh& back()`) with `PolygonMesh` as value type.
```

Comment:
You already said it's a model of `SequenceContainer`, there is no need to detail what it means (and `push_back()` would take an argument).